### DISCUSSION TITLE
A successful solution for packaging Nestjs

### DISCUSSION THREAD

**Discussion by shadowqcom** [oven-sh/bun#12253] [Ideas]
测试了一种打包Nestjs项目的方案：
1、使用ncc将nestjs项目及依赖提取出来：
```
npm i -g ncc
```
```
ncc build ./src/main.ts --minify
```
这将在dist目录输出完整版的项目文件，一个已经包含依赖的index.js文件和其他一些必要的文件。

2、使用bun打包 ‘被ncc打包后的文件’
```
bun build --compile --target=bun-linux-x64 ./dist/index.js --outfile myapp
```
这将生成一个二进制可执行文件。目前我测试在windows11 和 centos9上都正常使用。

测试时使用的是nestjs基础脚手架 没有任何业务代码。

**Comment by graffhyrum** [comment]
```sh
bun i -g @vercel/ncc
```
